sprintly/sprintly-kanban
# Add item port

#### What's this PR do?
Moves the add item functionality into the drawer design pattern we followed for the item detail page
Refactors multiple subitem components used across item card and add item into two `./component/subitems&subitem`

#### Where should the reviewer start?
`pages/add-item`

#### How should this be manually tested?
Open the app and visit `/add-item`
Under a story confirm that you can add subitems and remove them
Hit create and see that the subitems get batch created after

#### Background Context
I updated the event change emission signature on the product store to accept a `type && record` param so that the view component could look for passed arguments within its `_onChange` function.
Open to other suggestions/patterns. I failed to be able to find a way to return a reference to the newly created item via the Dispatcher otherwise.

I made the `subitems` component more generic so that its less context aware and can be repurposed across the

* item-card
* add-item
* item-detail on mobile

No attachment support yet.

#### Screenshots (if appropriate)
![image](https://cloud.githubusercontent.com/assets/2892213/9320941/1442e890-451d-11e5-96c6-1e99b3bc7500.png)

![image](https://cloud.githubusercontent.com/assets/2892213/9260835/1e9ded00-41bf-11e5-9ad4-34386b676f8c.png)
